timeline-chart/src/time-graph-model.ts

```diff
@@ -38,4 +39,8 @@ export namespace TimelineChart {
         data?: { [key: string]: any }
         // Q: Can I select an arrow?
     }
+
+    export interface TimeGraphAnnotation extends TimeGraphRowElementModel {
```

Comment:
Maybe this can be its own class (not extending), I don't think we need 'id' or 'selected' from the parent. Unless we support selecting an annotation. We don't need to have 'sourceId', it is the 'id' of the TimeGraphRowModel to which this annotation belongs.

We need 'type' (either CHART or TREE) to know if annotation is to be shown on the chart or in the tree (unless the tree is a widget external to timeline-chart? in which case only the CHART annotations would end up here). Alternatively, perhaps the 'range' can be optional, and if not present it means annotation goes on the tree.

Reply:
I think it's always Chart as tree is the way to go, since the tree is provided by theia.

Reply:
OK, the annotations with type 'TREE' will be filtered-out by the user and only those that are 'CHART' will be put in this model, so we don't need type.

I think we don't need sourceId, so perhaps we don't need a new class at all, annotations can fit in the TimeGraphRowElementModel. Unless it can be useful for 'instance of' checks?